apps/widgets/src/pages/admin-dashboard.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+import type { Widget, WidgetStatus } from '@walless/graphql';
+import { qlClient } from '@walless/graphql';
+import { updateWidgetStatus } from '@walless/graphql/mutation';
+import { allWidgets } from '@walless/graphql/query';
+import { Text } from '@walless/gui';
+import { HomeLayout } from 'components/layouts';
+import { WidgetInfo } from 'features/AdminDashboard';
+
+const fetchWidgetsData = async () => {
+	interface Data {
+		widgets: Widget[];
+	}
+	const data = (await qlClient.request(allWidgets)) as Data;
+	return data.widgets;
+};
+
+const AdminDashboardPage = () => {
+	const [widgets, setWidgets] = useState<Widget[]>([]);
+	const [isChanged, setIsChanged] = useState(false);
+
+	useEffect(() => {
+		try {
+			fetchWidgetsData().then(setWidgets);
+			setIsChanged(false);
+		} catch (error) {
+			console.error(error);
+		}
+	}, [isChanged]);
```

Comment:
as long as state migrated in-to Valtio, these fetchs (and its loading, error handlings) could be migrate as well it to its own Valtio actions, which keep component level simple and clean.